### ISSUE TITLE
Poor error message for computed string enums

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#25896]
https://twitter.com/larymagal/status/1021741263358443520

As a more basic example:

```ts
declare function bar(): string
enum E {
  foo = bar();
}
```

Currently we give the error

```
Type 'string' is not assignable to type 'E'.
```

However, we should give an error like

```
Only numeric enums can have computed members, but this expression has type 'string'.
  If you do not need exhaustiveness checks, consider using an object literal instead.
```

**Comment by mhegazy** [comment]
A PR would be appreciated.

**Comment by HarshKhatore** [comment]
@mhegazy I can try this?

**Comment by HarshKhatore** [comment]
Thanks, can you give me a direction on this?
I need to modify the Error handler message, so how do I move on to identifying the file? Is there a way to produce the error stack?

**Comment by mhegazy** [comment]
Error messages are defined in https://github.com/Microsoft/TypeScript/blob/master/src/compiler/diagnosticMessages.json. you will need to create a new error message, build (`jake local`), then use the new error message in https://github.com/Microsoft/TypeScript/blob/master/src/compiler/checker.ts

**Comment by amitdey1987** [comment]
@HarshKhatore are you still pursuing this. If not, then I would like to give it a try.

**Comment by DavidSotoA** [comment]
Can I try this one?

**Comment by DavidSotoA** [comment]
@mhegazy Hi, I try this but I need some help. I create a new error message in diagnosticMessages.json but I don't know wich function in checker I should put the new error message.

**Comment by awgaan** [comment]
Should this error cover this case?:
`  "Computed_values_are_not_permitted_in_an_enum_with_string_valued_members_2553" : "Computed values are not permitted in an enum with string valued members."`

**Comment by Codextor** [comment]
Sorry but I didn't notice there's already a PR. Wasted effort. :disappointed:

**Comment by zhongsp** [comment]
Hi @DanielRosenwasser , I've one question:

```ts
enum Foo {
  A = 5
}

enum Bar {
  B = Foo.A,
}
```

Is `Bar.B` a constant enum member or a computed enum member?

**Comment by DanielRosenwasser** [comment]
I'm not sure, I think constant but you'd need to check the compiler source.

**Comment by okmttdhr** [comment]
@DanielRosenwasser 
Hi, I submitted the PR https://github.com/microsoft/TypeScript/pull/37790, could you check it?
